src/averageAges.js

```diff
@@ -1,64 +1,57 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const FEMALE = 'f';
+const MALE = 'm';
+
+function sumAges(whos) {
+  return whos.reduce((accum, person) =>
+    (accum + (person['died'] - person['born'])), 0);
+};
+
+function getAgeDiff(whosChildren, whos) {
+  return ((whosChildren.reduce((accum, child) =>
+    accum + child['born'] - whos.find(mom => child['mother']
+      === mom['name'])['born'], 0)) / whosChildren.length).toFixed(2);
+};
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const allMen = people.filter(person => person['sex'] === MALE);
+
+  const centuryCadaversMen = century
+    ? allMen.filter(cadaver => Math.ceil(cadaver['died'] / 100) === century)
+    : false;
```

Comment:
instead of returning false you can return 'allMen' so later in return you don't need to check anything,  you can just return your sumAges function like 
`return sumAges(centuryCadaversMen) / centuryCadaversMen.length `

but in that case, would be better to change the 'centuryCadaversMen' to something else))